core/blockly.js

```diff
@@ -205,6 +205,13 @@ Blockly.svgResize = function(workspace) {
 // are multiple workspaces and non-main workspaces are able to accept input.
 Blockly.onKeyDown_ = function(e) {
   var mainWorkspace = Blockly.mainWorkspace;
+
+  if (mainWorkspace.options.readOnly &&
```

Comment:
I think you're conflicting a bit with the if statement immediate below. I'd actually move this below, take out the readOnly check in the other if so it reads like:

```
if (Blockly.utils.isTargetInput(e) || (...rendered && ... isVisible)) {
  ...
}
if (mainWorkspace.options.readOnly) {
  if (Blockly.keyboardAccessibilityMode) {
    Blockly.navigation.onKeyPress(e);
  }
  return; // Doesn't return a value
}
```

Reply:
fixed